### ISSUE TITLE
[turborepo] turbo prune ignore local peerDependencies

### ISSUE THREAD

**Issue by Naymi** [vercel/turbo#4915]
### What version of Turborepo are you using?

1.9.4

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Mac

### Describe the Bug

turbo prune ignore local peer deps

### Expected Behavior

turbo prune respects peerDependencies

### To Reproduce

```
git clone https://github.com/Naymi/turbo-prune-ignore-peer-demo
px turbo prune --scope depend-on-tested
```
out will only be depend-on-tested

### Reproduction Repo

https://github.com/Naymi/turbo-prune-ignore-peer-demo

**Comment by gravelg** [comment]
I am also being hit with this problem. I imagine this is due to the change in https://github.com/vercel/turbo/pull/4824

It seems that the lockfile generated by `turbo prune` is no longer usable by default when using `npm ci`, as npm expects to install peerDependencies by default [since npm 7](https://github.blog/2021-02-02-npm-7-is-now-generally-available/#peer-dependencies)

I've updated to pass the `--legacy-peer-deps` flag and it has fixed the issue on some repos but not all. Downgrading to turbo v1.9.3 fixes the problem.

**Comment by Naymi** [comment]
> Меня тоже мучает эта проблема. Я предполагаю, что это связано с изменением [# 4824](https://github.com/vercel/turbo/pull/4824)
> 
> Кажется, что файл блокировки, сгенерированный с помощью , `turbo prune`больше не используется по умолчанию при использовании `npm ci`, поскольку npm ожидает установки peerDependencies по умолчанию, [начиная с npm 7.](https://github.blog/2021-02-02-npm-7-is-now-generally-available/#peer-dependencies)
> 
> Я обновил, чтобы передать `--legacy-peer-deps`флаг, и это устранило проблему в некоторых рабочих областях, но не во всех. Переход на турбо версию 1.9.3 устраняет проблему.

Это не устранит проблему так как `peerDependencies` локальные

**Comment by Naymi** [comment]
> Меня тоже мучает эта проблема. Я предполагаю, что это связано с изменением [# 4824](https://github.com/vercel/turbo/pull/4824)
> 
> Кажется, что файл заблокирован, сгенерирован с помощью , `turbo prune`больше не используется по умолчанию при сборке `npm ci`, так как npm ожидает установки peerDependencies по умолчанию, [природа с npm 7.](https://github.blog/2021-02-02-npm-7-is-now-generally-available/#peer-dependencies)
> 
> Я обновил, чтобы передать `--legacy-peer-deps`, и это устранило проблему в некоторых рабочих группах, но не во всех. Переход на турбо версию 1.9.3 решает проблему.

This won't fix the problem since peerDependencies are local

**Comment by gravelg** [comment]
You are right, my issue is not exactly the same as yours. I'm having trouble with missing peerDependencies from (I believe) jest
```
#20 2.868 npm ERR! Missing: @swc/core@1.3.57 from lock file
#20 2.869 npm ERR! Missing: @swc/core-darwin-arm64@1.3.57 from lock file
#20 2.870 npm ERR! Missing: @swc/core-darwin-x64@1.3.57 from lock file
#20 2.871 npm ERR! Missing: @swc/core-linux-arm-gnueabihf@1.3.57 from lock file
#20 2.872 npm ERR! Missing: @swc/core-linux-arm64-gnu@1.3.57 from lock file
#20 2.872 npm ERR! Missing: @swc/core-linux-arm64-musl@1.3.57 from lock file
#20 2.874 npm ERR! Missing: @swc/core-linux-x64-gnu@1.3.57 from lock file
#20 2.875 npm ERR! Missing: @swc/core-linux-x64-musl@1.3.57 from lock file
#20 2.876 npm ERR! Missing: @swc/core-win32-arm64-msvc@1.3.57 from lock file
#20 2.877 npm ERR! Missing: @swc/core-win32-ia32-msvc@1.3.57 from lock file
#20 2.879 npm ERR! Missing: @swc/core-win32-x64-msvc@1.3.57 from lock file
#20 2.880 npm ERR! Missing: react@18.2.0 from lock file
#20 2.881 npm ERR! Missing: react-dom@18.2.0 from lock file
#20 2.882 npm ERR! Missing: scheduler@0.23.0 from lock file
```

**Comment by rskvazh** [comment]
Same problem. 1.9.4 prune peer dependencies and when `npm ci` I've got error: `Missing: svelte@3.59.1 from lock file`

**Comment by Naymi** [comment]
Still relevant!

**Comment by bernardobelchior** [comment]
Just faced this as well. Would be nice to have a fix!

**Comment by ljuborados** [comment]
We're facing the same issue.
We're stuck on 1.9.3 until this is fixed.

**Comment by attila** [comment]
This is still outstanding unfortunately.

We recently got bit by this and only a workaround helped where:
1. get the list of changed workspaces via `turbo run build --filter='...[base-branch]' --dry-run=json | jq -r '.packages | map(select(. != "//")) | join(" ")')`
2. take the list of local package names, and then check if any of them have local peers defined by parsing the lockfile
3. add any extra local packages names to the list as discovered above
4. call `turbo prune --out-dir=<path> <list of changed workspaces and their local peers>`
5. install dependencies then build the pruned repo as normal

Steps 2-3. contain our workaround to our monorepo of 100+ workspaces, that is only needed until the issue reported here is resolved.
